addon/components/nrg-datetime/calendar/component.js

```diff
@@ -167,19 +167,19 @@ export default Component.extend(EKMixin, EKFirstResponderOnFocusMixin, {
       const week = [];
       do {
         const date = calendar.date();
-        const monthIndex = calendar.month();
-        const isDifferentMonth = monthIndex !== this.selectedMonthIndex;
+        const month = calendar.month();
+        const year = calendar.year();
+        const isDifferentMonth = month !== this.selectedMonthIndex;
         const dateIsToday = calendar.isSame(today, 'date');
         const disabled = this._isDateDisabled(calendar, 'date') || isDifferentMonth;
         const selected = !disabled && this.selectedDayIndex === date;
 
         week.push({
           customClass: (dateIsToday && 'today') || '',
           display: date,
-          date,
+          date: { date, month, year },
```

Comment:
I added the year to the date because without it, selecting Dec. 31st 2019 while in Jan. 2020 results in Dec 31st 2020.